contrib/kafka/kafkaeventsource-operator/pkg/controller/kafkaeventsource/kafkaeventsource_controller.go

```diff
@@ -0,0 +1,335 @@
+
+	// Check the config - Environment Variables and annotations
+	if !reflect.DeepEqual(dep.Spec.Template, found.Spec.Template) {
+		found.Spec.Template = dep.Spec.Template
+		err := r.client.Update(context.TODO(), &found)
+		if err != nil {
+			log.Printf("failed to update Deployment configuration:, %s", err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Ensure the deployment size is the same as the spec. Defaults to 1 if not specified
+	size := kafkaEventSource.Spec.Replicas
+	if size == nil {
+		intendedSize := int32(1)
+		size = &intendedSize
+	}
+
+	if *found.Spec.Replicas != *size {
+		*found.Spec.Replicas = *size
+		log.Printf("Setting replicas:%d", *size)
+		err = r.client.Update(context.TODO(), &found)
+		if err != nil {
+			log.Printf("Failed to update Deployment: %s", found.Name)
+			return reconcile.Result{}, err
+		}
+		// Spec updated - return and requeue
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Update the KafkaEventSource status with the pod names
+	// List the pods for this event source's deployment
+	podList := &corev1.PodList{}
+	labelSelector := labels.SelectorFromSet(labelsForKafkaEventSource(kafkaEventSource.Name))
+	listOps := &client.ListOptions{Namespace: kafkaEventSource.Namespace, LabelSelector: labelSelector}
+	err = r.client.List(context.TODO(), listOps, podList)
+	if err != nil {
+		log.Printf("Failed to list pods when reconciling KafkaEventSource: %s", kafkaEventSource.Name)
+		return reconcile.Result{}, err
+	}
+
+	podNames := getPodNames(podList.Items)
+	nodes := kafkaEventSource.Status.Nodes
+	sort.Strings(podNames)
+	sort.Strings(nodes)
+
+	// Update status.Nodes if needed
+	if !reflect.DeepEqual(podNames, nodes) {
+		kafkaEventSource.Status.Nodes = podNames
+		log.Printf("Updating EventSource.Status.Nodes")
+		err := r.client.Update(context.TODO(), kafkaEventSource)
+		if err != nil {
+			log.Printf("failed to update KafkaEventSource status:, %s", err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// deploymentForKafka generates the Kubernetes deployment object for the EventSource
+func deploymentForKafka(kes *sourcesv1alpha1.KafkaEventSource) *appsv1.Deployment {
+	labels := labelsForKafkaEventSource(kes.Name)
+	replicas := kes.Spec.Replicas
+	envvars := getEnvVars(kes)
+
+	dep := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: kes.Name + "-",
+			Namespace:    kes.Namespace,
+			Labels:       labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: annotationsForKafkaEventSource(kes),
+					Labels:      labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Image:           "sjwoodman/kafkaeventsource:latest",
```

Comment:
instead of hardcoding here, we can pass the image as environment variable

Reply:
👍